src/Components/test/testassets/BasicTestApp/FileDownloaderComponent.razor

```diff
@@ -0,0 +1,33 @@
+@using Microsoft.AspNetCore.Components;
+
+@inject IFileDownloader FileDownload;
+
+<h1>File configuration</h1>
+File name:
+<br />
+@* Bind to fileName field *@
+<input type="text" @bind-value="@fileName" />
+<br />
+
+<h1>File download</h1>
+
+<button @onclick="SaveFile"> Save </button>
+
+@code {
+    private string fileName = "text.txt";
+
+    // Create file of random data from a new byte[] array
+    private Stream GetFileStream()
+    {
+        var randomBinaryData = new byte[50 * 1024];
```

Comment:
Let's make this `byte[]` contain repeated byte sequences from `0->255` (like we do in our file streaming/byte[] tests currently). Then in the tests, we can validate that all the expected data has been transferred. 

We'll also need to add E2E tests which leverage this component (ex. https://github.com/dotnet/aspnetcore/blob/main/src/Components/test/E2ETest/Tests/InputFileTest.cs)